src/Config/DatabasePrimitives/DatabaseObject.cs

```diff
@@ -290,6 +303,38 @@ public class ForeignKeyDefinition
     /// </summary>
     public List<string> ReferencingColumns { get; set; } = new();
 
+    public List<string> ResolveTargetColumns()
+    {
+        if (ReferencingEntityRole == RelationshipRole.Target)
+        {
+            return ReferencingColumns;
+        }
+        else if (ReferencedEntityRole == RelationshipRole.Target)
+        {
+            return ReferencedColumns;
+        }
+        else
+        {
+            throw new Exception("Unable to resolve target columns");
+        }
+    }
+
+    public List<string> ResolveSourceColumns()
+    {
+        if (ReferencingEntityRole == RelationshipRole.Source)
+        {
+            return ReferencingColumns;
+        }
+        else if (ReferencedEntityRole == RelationshipRole.Source)
+        {
+            return ReferencedColumns;
+        }
+        else
+        {
+            throw new Exception("Unable to resolve source columns");
```

Comment:
in this case, i think it's more accurate to say:
> message: "Unable to resolve source columns because this ForeignKeyDefinition relates a source entity and linking object.");

while the relationship in runtime config may denote the target columns, the specific ForeignKeyDefinition may only represent the relationship between the source entity and linking object.